### DISCUSSION TITLE
Do you really need to create a new token for every new repository you create?

### DISCUSSION THREAD

**Discussion by hkunz** [community/community#87198] [New to GitHub]
I had generated a classic token 2 weeks ago with literally all permissions and no expiration. And I used these steps to create a new repository.

Create empty repository on GitHub and then in terminal/console do the following:
```
cd path/to/repository-name
git init
git add .
git commit -m "Initial commit"
git remote add origin https://github.com/<user>/repository-name.git
git remote set-url origin https://<token>@github.com/<user>/repository-name.git
git push --set-upstream origin master
git push
```

Worked perfectly 2 weeks ago. Now my understanding is that I don't have to create a new token for every new repository I create but for some frustrating reason the `git push --set-upstream origin master` would not work for another new repository I created today. It either always asked for a password, which of course does not work anymore (no idea why it would still ask for passwords when tokens is the thing now), or it would immediately say Invalid username/password like:

```
$ git push --set-upstream origin master
remote: Invalid username or password.
fatal: Authentication failed for 'https://github.com/<user>/repository-name.git/'
```
or

```
git push --set-upstream origin master
Password for 'https://<token>@github.com':
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/<user>/repository-name.git/'
```

So I tried and tried until I decided this token stuff didn't work anymore so I decided to create a second token today. I now have 2 tokens: The old 2-week-old token which didn't work for the new repository, and now my second token with the same all permissions and no expiration and did literally the same steps again. And it worked with the new token. What gives? Is this a bug or do I really need to create a new token whenever I create a new repository?

**Comment by queenofcorgis** [comment]
👋🏼 You should be able to access every repository with one token. 

Are you able to check in your settings the scope of this token? Use this doc for [classic personal access tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic) and this for [fine-grained](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token).

**Comment by MuthuKumar-50** [comment]
How did you solve that issues

**Comment by hkunz** [comment]
@MuthuKumar-50 I just created another a new one. problem solved lol

**Comment by Chamal1120** [comment]
why use tokens?
you can easily get the github cli and login with your account. That way you don't need tokens anymore.

**Reply by Charan-N** [reply]
Not happening anymore, I am getting a message saying password authentication removed on Aug13 2021 on my personal laptop. However, I was able to login easily on my work laptop. IDK what to do.
